### ISSUE TITLE
[BUG] no lexer for filename 'xxx' found

### ISSUE THREAD

**Issue by ewels** [willmcgugan/rich#996]
**Describe the bug**
At the top of the tracebacks for my tool on exceptions I'm getting the following (see [this GitHub Actions workflow](https://github.com/nf-core/tools/pull/821/checks#step:6:224)):

```
╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /opt/hostedtoolcache/Python/3.7.9/x64/bin/nf-core:8 in <module>              │
│                                                                              │
│ no lexer for filename 'nf-core' found                                        │
```

Alternatively, if testing locally:

```
╭─────────────────────── Traceback (most recent call last) ───────────────────────╮
│ /Users/ewels/miniconda3/envs/py3/bin/nf-core:33 in <module>                     │
│                                                                                 │
│ no lexer for filename 'nf-core' found                                           │
```

Syntax highlighting in the traceback does work correctly locally.

**To Reproduce**
Any traceback from https://github.com/nf-core/tools has this warning. To test locally I added an `assert False` statement [here](https://github.com/nf-core/tools/blob/8c68650409a3f95376dd55c291dae5f9fc346347/nf_core/__main__.py#L150) and ran `nf-core list`.

**Platform**
Linux (GitHub Actions), Mac OSX (locally).

**Diagnose**

<details>

<summary><code>python -m rich.diagnose</code></summary>

```
❯ python -m rich.diagnose
╭────────────────────────────────────────────────────────── <class 'rich.console.Console'> ───────────────────────────────────────────────────────────╮
│ A high level console interface.                                                                                                                     │
│                                                                                                                                                     │
│ ╭─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮ │
│ │ <console width=151 ColorSystem.TRUECOLOR>                                                                                                       │ │
│ ╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                                                                     │
│     color_system = 'truecolor'                                                                                                                      │
│         encoding = 'utf-8'                                                                                                                          │
│             file = <_io.TextIOWrapper name='<stdout>' mode='w' encoding='UTF-8'>                                                                    │
│ is_dumb_terminal = False                                                                                                                            │
│       is_jupyter = False                                                                                                                            │
│      is_terminal = True                                                                                                                             │
│   legacy_windows = False                                                                                                                            │
│         no_color = False                                                                                                                            │
│          options = ConsoleOptions(legacy_windows=False, min_width=1, max_width=151, is_terminal=True, encoding='utf-8', justify=None,               │
│                    overflow=None, no_wrap=False, highlight=None)                                                                                    │
│           record = False                                                                                                                            │
│         safe_box = True                                                                                                                             │
│             size = ConsoleDimensions(width=151, height=43)                                                                                          │
│        soft_wrap = False                                                                                                                            │
│           stderr = False                                                                                                                            │
│            style = None                                                                                                                             │
│         tab_size = 8                                                                                                                                │
│            width = 151                                                                                                                              │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

</details>

<details>

<summary><code>python -m rich._windows</code></summary>

```
 ❯ python -m rich._windows
platform="Darwin"
WindowsConsoleFeatures(vt=False, truecolor=False)
```

</details>

<details>

<summary><code>pip freeze | grep rich</code></summary>

```
 ❯ pip freeze | grep rich
rich==9.10.0
```

</details>

**Comment by willmcgugan** [comment]
It's because the `bin/nf-core` file doesn't have a .py extension. I could just assume Python, which will be correct 99.9% of the time.

**Comment by ewels** [comment]
I wondered if that was the case. The file in question is auto-generated due to the [`console_scripts` `entry_point`](https://github.com/nf-core/tools/blob/8c68650409a3f95376dd55c291dae5f9fc346347/setup.py#L32):

```python
#!/Users/ewels/miniconda3/envs/py3/bin/python3.7
# EASY-INSTALL-ENTRY-SCRIPT: 'nf-core','console_scripts','nf-core'
import re
import sys

# for compatibility with easy_install; see #2198
__requires__ = 'nf-core'

try:
    from importlib.metadata import distribution
except ImportError:
    try:
        from importlib_metadata import distribution
    except ImportError:
        from pkg_resources import load_entry_point


def importlib_load_entry_point(spec, group, name):
    dist_name, _, _ = spec.partition('==')
    matches = (
        entry_point
        for entry_point in distribution(dist_name).entry_points
        if entry_point.group == group and entry_point.name == name
    )
    return next(matches).load()


globals().setdefault('load_entry_point', importlib_load_entry_point)


if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(load_entry_point('nf-core', 'console_scripts', 'nf-core')())

```

I guess you could also look at the file hashbang if present? But probably not super precise either.

**Comment by willmcgugan** [comment]
> I guess you could also look at the file hashbang if present? But probably not super precise either.

True, if the first line starts with hash bang and contains the text "python" assume a Python lexer, otherwise plain text. Seems like a reasonable approach.

**Comment by willmcgugan** [comment]
Should be fixed in 9.11.0

**Comment by ewels** [comment]
Tested, working great in my hands too. Brilliant stuff - thank you!
